html5/js/Client.js

```diff
@@ -1651,8 +1651,8 @@ class XpraClient {
       return true;
     }
 
-    if (this.server_readonly || !this.connected) {
-      return window == undefined;
+    if (this.server_readonly || !this.connected || (!win && this.server_is_shadow)) {
```

Comment:
This feels convoluted and I don't understand why it's here just by reading the code.
Can you separate it out and add a comment explaining why it's needed or useful?